client/signup/steps/woocommerce-install/step-store-address/index.tsx

```diff
@@ -162,3 +212,56 @@ export default function StepStoreAddress( props: WooCommerceInstallProps ): Reac
 		/>
 	);
 }
+
+function ControlError( props: { error: string } ): ReactElement | null {
+	const { error } = props;
+	if ( error ) {
+		return <FormInputValidation isError={ true } isValid={ false } text={ error } />;
+	}
+	return null;
+}
+
+function useAddressFormValidation( siteId: number ) {
+	const { get } = useSiteSettings( siteId );
+	const { __ } = useI18n();
+
+	const [ errors, setErrors ] = useState( {} as Record< optionNameType, string > );
+
+	const validate = () => {
+		errors[ WOOCOMMERCE_STORE_ADDRESS_1 ] = ! get( WOOCOMMERCE_STORE_ADDRESS_1 )
+			? __( 'Please add an address' )
+			: '';
+		errors[ WOOCOMMERCE_STORE_ADDRESS_2 ] = ''; // Optional field
+		errors[ WOOCOMMERCE_DEFAULT_COUNTRY ] = ! get( WOOCOMMERCE_DEFAULT_COUNTRY )
+			? __( 'Please select a country / region' )
+			: '';
+		errors[ WOOCOMMERCE_STORE_CITY ] = ! get( WOOCOMMERCE_STORE_CITY )
+			? __( 'Please add a city' )
+			: '';
+		errors[ WOOCOMMERCE_STORE_POSTCODE ] = ! get( WOOCOMMERCE_STORE_POSTCODE )
+			? __( 'Please add a postcode' )
+			: '';
+		errors[ WOOCOMMERCE_ONBOARDING_PROFILE ] = ! emailValidator.validate(
+			get( WOOCOMMERCE_ONBOARDING_PROFILE )?.[ 'store_email' ]
+		)
+			? __( 'Please add a store email' )
```

Comment:
Please ignore if not relevant. but is "store email" a thing? 

What about a more generic `Please add a valid email address`?

Reply:
Yea, I think you're right, I'll do this in a follow up. I think the error message positioning needs more tweaking anyway.